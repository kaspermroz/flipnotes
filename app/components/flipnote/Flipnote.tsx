import { useState } from "react";
import { Box, Center, Text } from "@chakra-ui/react";

type FlipnoteProps = {
  title: string;
  content: string;
};

export const Flipnote = ({ title, content }: FlipnoteProps) => {
  const [isFlipped, setIsFlipped] = useState(false);

  return (
    <Box
      className="flip-card"
      onClick={() => setIsFlipped(!isFlipped)}
      w={700}
      h={700}
      borderRadius={25}
    >
      <Box
        className={`flip-card-inner ${
          isFlipped ? "flip-card-inner-rotate" : ""
        }`}
        borderRadius={25}
      >
        <Box
          className="flip-card-front"
          borderRadius={25}
          bg="#FFFFF0"
          border="1px solid #F6E05E"
          p={4}
        >
          <Center h="full">
            <Text fontSize="4xl">
              <b>{title}</b>
            </Text>
          </Center>
        </Box>
        <Box
          className="flip-card-back"
          borderRadius={25}
          bg="#FFFFF0"
          border="1px solid #F6E05E"
          p={4}
        >
          <Center h="full">
            <Text fontSize="4xl">{content}</Text>
          </Center>
        </Box>
      </Box>
    </Box>
  );
};
