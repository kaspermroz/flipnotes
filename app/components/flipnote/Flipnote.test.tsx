import React from "react";
import { render, screen } from "@testing-library/react";
import { Flipnote } from "./Flipnote";

const defaultProps = {
  title: "test title",
  content: "test content",
};

describe("<Flipnote />", () => {
  it("should render", () => {
    const { container } = render(<Flipnote {...defaultProps} />);
    expect(container).toBeDefined();
  });

  it("should flip after click", async () => {
    const { container } = render(<Flipnote {...defaultProps} />);
    expect(container).toBeDefined();

    const flipnote = screen.getByTestId("flipnote-container");
    expect(flipnote).toBeInTheDocument();

    const flipnoteFront = screen.getByTestId("flipnote-front");
    expect(flipnoteFront).toBeInTheDocument();
    expect(flipnoteFront).toHaveTextContent(defaultProps.title);

    const flipnoteBack = screen.getByTestId("flipnote-back");
    expect(flipnoteBack).toBeInTheDocument();
    expect(flipnoteBack).toHaveTextContent(defaultProps.content);
    expect(container).toMatchSnapshot();

    await flipnote.click();
    expect(flipnoteFront).toHaveTextContent(defaultProps.title);
    expect(flipnoteBack).toHaveTextContent(defaultProps.content);
    expect(container).toMatchSnapshot();
  });
});
