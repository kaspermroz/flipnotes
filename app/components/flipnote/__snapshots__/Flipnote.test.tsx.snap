// Vitest Snapshot v1

exports[`<Flipnote /> > should flip after click 1`] = `
<div>
  <div
    class="flip-card css-1nx0mta"
    data-testid="flipnote-container"
  >
    <div
      class="flip-card-inner  css-xcat2f"
    >
      <div
        class="flip-card-front css-2n4rnq"
        data-testid="flipnote-front"
      >
        <div
          class="css-1yua5i9"
        >
          <p
            class="chakra-text css-aq7y69"
          >
            <b>
              test title
            </b>
          </p>
        </div>
      </div>
      <div
        class="flip-card-back css-2n4rnq"
        data-testid="flipnote-back"
      >
        <div
          class="css-1yua5i9"
        >
          <p
            class="chakra-text css-aq7y69"
          >
            test content
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Flipnote /> > should flip after click 2`] = `
<div>
  <div
    class="flip-card css-1nx0mta"
    data-testid="flipnote-container"
  >
    <div
      class="flip-card-inner flip-card-inner-rotate css-xcat2f"
    >
      <div
        class="flip-card-front css-2n4rnq"
        data-testid="flipnote-front"
      >
        <div
          class="css-1yua5i9"
        >
          <p
            class="chakra-text css-aq7y69"
          >
            <b>
              test title
            </b>
          </p>
        </div>
      </div>
      <div
        class="flip-card-back css-2n4rnq"
        data-testid="flipnote-back"
      >
        <div
          class="css-1yua5i9"
        >
          <p
            class="chakra-text css-aq7y69"
          >
            test content
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
